### ISSUE TITLE
False positives of UnusedPrivateMember in operator overloads

### ISSUE THREAD

**Issue by sergeys-opera** [detekt/detekt#3565]
My issue seems to be exactly like https://github.com/detekt/detekt/issues/3096 which marked as fixed. Maybe a regression was introduced in this rule? For me **detekt** fails on a somewhat trivial code with operator overloads (see the code snippet below).

## Expected Behavior
Detekt doesn't report **UnusedPrivateMember** for operator overloads in use.

## Observed Behavior
Detekt reports **UnusedPrivateMember** for operator overloads in use.

## Steps to Reproduce
The failing code is:
```
private operator fun PointF.times(factor: Float) = PointF(factor * x, factor * y)
private operator fun Float.times(r: PointF) = r * this
```
Detekt flags `PointF.times` as unused but it's clearly used by the operator fun one line below it and in some other places via `*`.
Removing this method breaks the compilation.

## Your Environment
* Version of detekt used: 1.16.0
* Version of Gradle used (if applicable): 6.8
* Operating System and version: https://termbin.com/tp40

**Comment by marschwar** [comment]
Hi @sergeys-opera ,

thank your for pointing this out. I will look into it.

**Comment by marschwar** [comment]
I was able to reproduce the behavior by running detekt without [type resolution](https://detekt.github.io/detekt/type-resolution.html). Only with type resolution detekt can determine that the functions are actually used and it does so correctly.

**Comment by cortinico** [comment]
Sadly another occurrence of #2994 

I'd say we can close this. @sergeys-opera Can you please rerun detekt with type resolution (e.g. `detektMain` task) and confirm that the rule works correctly?

**Comment by schalkms** [comment]
Agreed. I think we should close this then.
